hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AbfsConfiguration.java

```diff
@@ -474,15 +474,20 @@ public class AbfsConfiguration{
       FS_AZURE_APACHE_HTTP_CLIENT_MAX_IO_EXCEPTION_RETRIES, DefaultValue = DEFAULT_APACHE_HTTP_CLIENT_MAX_IO_EXCEPTION_RETRIES)
   private int maxApacheHttpClientIoExceptionsRetries;
 
-  /**
-   * Max idle TTL configuration for connection given in
-   * {@value org.apache.hadoop.fs.azurebfs.constants.ConfigurationKeys#FS_AZURE_APACHE_HTTP_CLIENT_IDLE_CONNECTION_TTL}
-   * with default of
-   * {@value org.apache.hadoop.fs.azurebfs.constants.FileSystemConfigurations#DEFAULT_HTTP_CLIENT_CONN_MAX_IDLE_TIME}
-   */
-  @LongConfigurationValidatorAnnotation(ConfigurationKey = FS_AZURE_APACHE_HTTP_CLIENT_IDLE_CONNECTION_TTL,
-      DefaultValue = DEFAULT_HTTP_CLIENT_CONN_MAX_IDLE_TIME)
-  private long maxApacheHttpClientConnectionIdleTime;
+  @IntegerConfigurationValidatorAnnotation(ConfigurationKey =
+      FS_AZURE_APACHE_HTTP_CLIENT_MAX_CACHE_CONNECTION_SIZE, DefaultValue = DEFAULT_HTTP_CLIENT_CONN_MAX_CACHED_CONNECTIONS,
+      MinValue = MIN_HTTP_CLIENT_CONN_MAX_CACHED_CONNECTIONS, MaxValue = MAX_HTTP_CLIENT_CONN_MAX_CACHED_CONNECTIONS)
+  private int maxApacheHttpClientCacheConnections;
+
+  @IntegerConfigurationValidatorAnnotation(ConfigurationKey =
+      FS_AZURE_CACHE_WARMUP_CONNECTION_COUNT, DefaultValue = DEFAULT_APACHE_CACHE_WARMUP_CONNECTION_COUNT,
+      MinValue = 0, MaxValue = MIN_HTTP_CLIENT_CONN_MAX_CACHED_CONNECTIONS)
```

Comment:
should we keep a separate variable for this as the max value configured by min of something is a bit misleading or are the two always interdependent ?